phoenix-ios/phoenix-ios/officers/BusinessManager.swift

```diff
@@ -65,6 +65,12 @@ class BusinessManager {
 	///
 	public let canMergeChannelsForSplicingPublisher = CurrentValueSubject<Bool, Never>(false)
 	
+	/// Reports whether or not the notifySrvExt process is running, and connected to the Peer.
+	/// When this is true, we (the main app process) are prevented from connecting to the Peer
+	/// (otherwise doing so would force-disconnect the notifySrvExt, which may be processing an incoming payment)
+	///
+	public let srvExtConnectedToPeer = CurrentValueSubject<Bool, Never>(false)
```

Comment:
I'm concerned there will be inevitable edge cases/race conditions where this flag would end up always preventing the peer to connect. Is there a timeout so that we eventually always end up on our feet?

Reply:
Yes. The way it works internally is:
* after the notifySrvExt connects to the peer, it stores a timestamp in the shared UserDefaults
* it updates this timestamp every 2 seconds while it's alive & running
* the main app listens to this property, and assumes the notifySrvExt may still be alive if the timestamp is within the last 5 seconds

So we're not dependent on the notifySrvExt completing properly and performing some clean-up operation. It can simply crash, and the main app will recover gracefully.